benchexec/tools/metaval.py

```diff
@@ -126,7 +137,7 @@ def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
                 ]
                 + additionalPathArgument
                 + witnessTypeArgument
-                + tasks
+                + list(task.input_files_or_empty)
```

Comment:
Probably `task.input_files` or even `task.single_input_file`? Does MetaVal handle more than one input file?

Reply:
This really is overengineering on the benchexec side. Why not throw everything we have on the tools and let them deal with it?

But OK I will change it to task.single_input_file

Reply:
Well, it gives nicer error messages to the user. But nobody is forced to use this, so I don't see why this is over-engineering? I just don't see a reason to use the longer `task.input_files_or_empty` if a shorter method will be easier to understand and provide better user experience?

Reply:
> Well, it gives nicer error messages to the user

But it adds too much complexity just for an error message. The interface of `Task` would be pretty clear, but it is overly complicated with all those `@property` methods.

The whole benchexec tool-info module thing is over-engineered. A uniform wrapper around each tool with a clear command line interface would be way better if you ask me (I know you will probably argue that the tool-info modules are exactly that, but I disagree). There are a lot of YAGNIs and WETs involved in this tool-info module business. It is way too bloaty for what it is actually doing. It also means we have to develop both in the tool repository and at the tool-info module level. Functionality can be shifted between these two. tool-info modules need to disappear if you ask me.

If you want to continue this discussion, please let's not do it here. I implemented the change you asked for so this thread should be resolved.